examples/cluster/plot_linkage_comparison.py

```diff
@@ -0,0 +1,148 @@
+"""
+================================================================
+Comparing different hierarchical linkage methods on toy datasets
+================================================================
+
+This example shows characteristics of different linkage
+methods for hierarchical clustering on datasets that are
+"interesting" but still in 2D.
+
+The main observations to make are:
+
+- single linkage is fast, and can perform well on
+  non-globular data, but it performs poorly in the
+  presence of noise.
+- average and complete linkage perform well on
+  cleanly separated globular clusters, but have mixed
+  results otherwise.
+- Ward is the most effective method for noisy data.
+
+While these examples give some intuition about the
+algorithms, this intuition might not apply to very high
+dimensional data.
+"""
+print(__doc__)
+
+import time
+import warnings
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn import cluster, datasets
+from sklearn.preprocessing import StandardScaler
+from itertools import cycle, islice
+
+np.random.seed(0)
+
+# ============
+# Generate datasets. We choose the size big enough to see the scalability
+# of the algorithms, but not too big to avoid too long running times
+# ============
```

Comment:
Should we rather use sphinx-gallery separators here: insert a line of continuous "#" longer than 70 chars before the block and only before. Sphinx-gallery will use this to create html rendering and Jupyter notebooks.